### PR TITLE
chore: Add some async improvements

### PR DESCRIPTION
**Pull request by Lash-L**
Relates to #228

### CODE REVIEW

**Comment by humbertogontijo** [comment]
I think it would be best to just keep it simple and create a counter to be used as the request_id. What you think?

**Comment by humbertogontijo** [comment]
I don't know if it can conflict between cloud and local. But we can do something like top to bottom on cloud and bottom to top on local

**Comment by Lash-L** [comment]
> I don't know if it can conflict between cloud and local. But we can do something like top to bottom on cloud and bottom to top on local

Sorry, I misspoke, I was specifically talking about commands run via the roborock app. i.e. I do something on my phone's app, it happens to be request id 34212 and that happens to be the next sequential id for HA. But it is probably rare enough that that edge case shouldn't even be really considered.

**Comment by Lash-L** [comment]
Okay made the changes, what do you think about that?

**Review by humbertogontijo — APPROVED** [review]

File: roborock/api.py [hunk 1]
```diff
@@ -113,6 +113,13 @@ def _async_response(
         self, request_id: int, protocol_id: int = 0
     ) -> Coroutine[Any, Any, tuple[Any, VacuumError | None]]:
         queue = RoborockFuture(protocol_id)
+        if request_id in self._waiting_queue:
+            new_id = get_next_int(10000, 32767)
+            _LOGGER.warning(
+                f"Attempting to create a future with an existing request_id... New id is {new_id}. "
+                f"Code may not function properly."
+            )
+            request_id = new_id
```
Comment (allenporter) [inline roborock/api.py:122]: I think this ends up writing the future with a different key, but does not change the request id chosen by the caller. I think this means the response will still be left hanging and/or associated with the wrong request?

File: roborock/roborock_message.py [hunk 2]
```diff
@@ -155,9 +155,9 @@ class MessageRetry:
 class RoborockMessage:
     protocol: RoborockMessageProtocol
     payload: bytes | None = None
-    seq: int = randint(100000, 999999)
+    seq: int = get_next_int(100000, 999999)
```
Comment (allenporter) [inline roborock/roborock_message.py:158]: I believe these numbers are set once statically, and not once per message creation, which i think was the intent.  I think this is the equivalent of setting a constant for all created messages.

Comment (Lash-L) [reply roborock/roborock_message.py:158]: Looks like you may be correct? I modified what was existing prior without thinking about it much, I thought data classes worked a bit like init but obviously not. Good catch

**Pull request merged**